src/compiler/sourcemap.ts

```diff
@@ -333,12 +336,21 @@ namespace ts {
                 const oldSource = currentSource;
                 if (source === oldSource) source = undefined;
 
+                let nameIndex;
+                if (isIdentifier(node)) {

```

Comment:
We shouldn't need to record a name for identifiers that haven't changed. `isParseTreeNode` can be used to test for this case.

Reply:
Most of source map producers (e.g. uglify) output non-changed names in the map anyway.